src/tools/compiletest/src/runtest.rs

```diff
@@ -1816,12 +1826,22 @@ impl<'test> TestCx<'test> {
         }
     }
 
-    fn dump_output(&self, out: &str, err: &str) {
+    fn dump_output(&self, print_output: bool, proc_name: &str, out: &str, err: &str) {
         let revision = if let Some(r) = self.revision { format!("{}.", r) } else { String::new() };
 
         self.dump_output_file(out, &format!("{}out", revision));
         self.dump_output_file(err, &format!("{}err", revision));
-        self.maybe_dump_to_stdout(out, err);
+
+        if !print_output {
+            return;
+        }
+
+        let proc_name = Path::new(proc_name).file_name().unwrap().to_string_lossy();
+        eprintln!("------{proc_name} stdout------------------------------");
```

Comment:
Discussion: there is also a slight issue that we have to dump somewhere. stderr is no-go because it will interleave with stdout from libtest. stdout will *also* interleave with libtest test status dots, e.g.

```
------------------------------------------
.ii------rmake.exe stdout------------------------------

------rmake.exe stderr------------------------------

------------------------------------------
  88/382
.i------rmake.exe stdout------------------------------

------rmake.exe stderr------------------------------

------------------------------------------
.------rmake.exe stdout------------------------------

------rmake.exe stderr------------------------------

------------------------------------------
.------rmake.exe stdout------------------------------

------rmake.exe stderr------------------------------

------------------------------------------
.
Build completed unsuccessfully in 0:00:40
```

Not *great*, but I'm okay with accepting this as-is, unless we can force libtest to verbose list test names instead.

Reply:
i don't think we should touch this logic at all. the whole point of --nocapture is that i don't trust libtest to decide which output to print for me. if you want pretty output we should teach compiletest to respect `--show-output` instead, which in normal libtest prints the output at the end:
```
; c t --lib -- --show-output
warning: `[project]` is deprecated in favor of `[package]`
    Finished `test` profile [unoptimized + debuginfo] target(s) in 0.01s
     Running unittests src/lib.rs (/home/jyn/.local/lib/cargo/target/debug/deps/example-d78220620848f9e6)

running 2 tests
test bar ... ok
test foo ... ok

successes:

---- bar stdout ----
jieyouxu

---- foo stdout ----
hi


successes:
    bar
    foo

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
```